--- libgestures/src/gestures.cc	2015-09-14 22:16:46.676359587 -0600
+++ libgestures.new/src/gestures.cc	2015-09-14 14:29:34.079842209 -0600
@@ -32,6 +32,7 @@
 #include "gestures/include/split_correcting_filter_interpreter.h"
 #include "gestures/include/string_util.h"
 #include "gestures/include/stuck_button_inhibitor_filter_interpreter.h"
+#include "gestures/include/fling_to_scroll_filter_interpreter.h"
 #include "gestures/include/t5r2_correcting_filter_interpreter.h"
 #include "gestures/include/trace_marker.h"
 #include "gestures/include/tracer.h"
@@ -502,8 +503,8 @@
       return;
     }
   }
-
   Interpreter* temp = new ImmediateInterpreter(prop_reg_.get(), tracer_.get());
+  temp = new FlingToScrollFilterInterpreter(prop_reg_.get(), temp, tracer_.get());  
   temp = new FlingStopFilterInterpreter(prop_reg_.get(), temp, tracer_.get());
   temp = new ClickWiggleFilterInterpreter(prop_reg_.get(), temp, tracer_.get());
   temp = new PalmClassifyingFilterInterpreter(prop_reg_.get(), temp,
@@ -539,6 +540,7 @@
 
 void GestureInterpreter::InitializeTouchpad2(void) {
   Interpreter* temp = new ImmediateInterpreter(prop_reg_.get(), tracer_.get());
+  temp = new FlingToScrollFilterInterpreter(prop_reg_.get(), temp, tracer_.get());  
   temp = new FlingStopFilterInterpreter(prop_reg_.get(), temp, tracer_.get());
   temp = new ClickWiggleFilterInterpreter(prop_reg_.get(), temp, tracer_.get());
   temp = new PalmClassifyingFilterInterpreter(prop_reg_.get(), temp,
