--- gestures/src/scaling_filter_interpreter.cc	2015-03-11 11:56:04.805543112 -0600
+++ gestures.new/src/scaling_filter_interpreter.cc	2015-03-15 20:36:18.595170108 -0600
@@ -280,13 +280,16 @@
                                           Metrics* metrics,
                                           MetricsProperties* mprops,
                                           GestureConsumer* consumer) {
+  float extra_screen_scale = 1.6;
+  float base_dpi = 96.0;
+
   tp_x_scale_ = 1.0 / hwprops->res_x;
   tp_y_scale_ = 1.0 / hwprops->res_y;
   tp_x_translate_ = -1.0 * (hwprops->left * tp_x_scale_);
   tp_y_translate_ = -1.0 * (hwprops->top * tp_y_scale_);
 
-  screen_x_scale_ = hwprops->screen_x_dpi / 25.4;
-  screen_y_scale_ = hwprops->screen_y_dpi / 25.4;
+  screen_x_scale_ = (hwprops->screen_x_dpi / base_dpi) * extra_screen_scale;
+  screen_y_scale_ = (hwprops->screen_y_dpi / base_dpi) * extra_screen_scale;
 
   if (hwprops->orientation_maximum)
     orientation_scale_ =
